src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -390,12 +390,12 @@ public void setContainerCapStr(String containerCapStr) {
     }
 
     public void setContainerCap(Integer containerCap) {
-        this.containerCap = (containerCap != null && containerCap >= 0) ? containerCap : null;
+        this.containerCap = (containerCap != null && containerCap > 0) ? containerCap : null;
     }
 
     public String getContainerCapStr() {
-        // serialized Integer.MAX_VALUE means no limit
-        return (containerCap == null || containerCap == Integer.MAX_VALUE) ? "" : String.valueOf(containerCap);
+        // null, serialized Integer.MAX_VALUE, or 0 means no limit
+        return (containerCap == null || containerCap == Integer.MAX_VALUE || containerCap == 0) ? "" : String.valueOf(containerCap);
```

Comment:
Ineffective because `mayAddProvisionedSlave` is calling `getContainerCap`, not this method, so it sees 0 and refuses to provision pods.